ToolChain/Assembler/Parser/Parser.py

```diff
@@ -162,239 +179,128 @@ def _parseCodeLine(self, line: list=None):
         :param line:
         :return:
         """
-        lineInstruction = None
 
-        buildInstruction = Instruction(skipValidation=True)  # This is used as model in early instruction building
+        lineInstruction = None
+        buildInstruction = Instruction(skipValidation=True)  # This is used as a model in early instruction building
         finalCode = 0x00
         form = None
         possibleCodes = None
 
+        # We evaluate the line and extract any bits of relevant info to build the appropriate instruction
         self._evaluateAndExtractInfoFromLine(line, buildInstruction)
 
         if type(buildInstruction.instructionCode) is not str:
-            # Instruction is real not just memory address or comment
+
+            # We assume it's an instruction (integer), since memory references or comments would be strings
+            # Parser will verify all possible codes, and generate the appropriate instruction based on its form
             possibleCodes = self._findPossibleCodesForInstruction(buildInstruction)
             instructionCode, instructionForm = self._getInstructionCodeAndFormUsingPossibleCodes(buildInstruction,
                                                                                                  possibleCodes)
+
             if instructionForm is None or instructionCode is None:
+                # Code evaluated in function isn't valid
                 raise ValueError("Assembler general error")
```

Comment:
Comment, code, and error message don't match. 
If anything, the line of code couldn't be recognized. Not sure what edge case triggers this. Would be nice to verify if this will cause it to print the line from the input file. I know some do.